app/models/maintenance_tasks/runner.rb

```diff
@@ -54,11 +54,15 @@ def run(name:, csv_file: nil, arguments: {})
 
       run.enqueued!
       enqueue(run)
-      Task.named(name)
+      returns(run)
     end
 
     private
 
+    def returns(run)
```

Comment:
If we want this to be overridable / documented, we should probably make this public right?

Reply:
I don't really want to make it public. Just make it easier to return a Run in Shopify/shopify, without having to re-write the full method, or introducing a flag/API.

But I think this is not enough though because overriding `returns` as it is would break the controller and CLI. I have to find another way, maybe I need to add a flag to `Runner.run` after all…

Reply:
It may also be that this is the wrong time to upstream. Especially if

> `Run` is a private API that most apps shouldn't tinker with.

We may be able to make a better decision when we have two instances of needing to return a run, if that exists in our use-case.

Reply:
Agree that since `Run` is private API it doesn't make sense to change the method to return the `Run` in the general case.

> as it is would break the controller and CLI.

Is this really a problem? The only thing talking to the gem's `Runner` in Core will be our CPT that handles enqueuing - we'll have a top-level runner (and separate CLI / controller entrypoints) so I don't think modifying this return value will cause issues unless I'm missing something?

Reply:
Ahh nice!! Yep, even better without having to monkey patch 🙌

Reply:
Yeah, agree yield was a good choice 👍